rules/spec-only.js

```diff
@@ -35,14 +37,19 @@ module.exports = {
       MemberExpression(node) {
         if (
           node.object.type === 'Identifier' &&
+          (!node.computed || node.property.type === 'Literal') &&
           node.object.name === 'Promise' &&
-          !(node.property.name in PROMISE_STATICS) &&
+          ((node.property.name && !PROMISE_STATICS.has(node.property.name)) ||
+            (node.property.value &&
+              !PROMISE_STATICS.has(node.property.value))) &&
+          (node.property.name !== 'prototype' ||
+            !PROMISE_INSTANCE_PROPS.has(node?.parent?.property?.name)) &&
```

Comment:
In the future, to `Promise.prototype` could be added new methods, like it was with `.finally`. And in this case here will be a situation like with static `Promise.try` now - it's still not in the spec, but it's on stage 3 and shipped in the latest Chrome, for usage it should be added to `allowedMethods`. But here is no similar check for instance methods.

Reply:
Thanks, I've added an instance method check.